core/node/node_sync/src/batch_status_updater/mod.rs

```diff
@@ -98,6 +99,377 @@ impl MainNodeClient for Box<DynClient<L2>> {
+        pool: ConnectionPool<Core>,
+        sl_chain_id: SLChainId,
+    ) -> Self {
+        Self {
+            sl_client,
+            diamond_proxy_addr,
+            contract: zksync_contracts::hyperchain_contract(),
+            pool,
+            sl_chain_id,
+        }
+    }
+
+    async fn validate_commit_tx_against_db(
+        &self,
+        commit_tx_hash: H256,
+        batch_number: L1BatchNumber,
+    ) -> anyhow::Result<L1TransactionValidationResult> {
+        let db_batch = match self
+            .pool
+            .connection_tagged("sync_layer")
+            .await?
+            .blocks_dal()
+            .get_l1_batch_metadata(batch_number)
+            .await?
+        {
+            Some(batch) => batch,
+            None => {
+                tracing::debug!("Batch {} is not found in the database. Cannot verify commit transaction right now", batch_number);
+                return Ok(L1TransactionValidationResult::TryLater);
+            }
+        };
+
+        let receipt = self
+            .sl_client
+            .tx_receipt(commit_tx_hash)
+            .await?
+            .context("Failed to fetch commit transaction receipt from SL")?;
+        if receipt.status != Some(U64::one()) {
+            let err = anyhow::anyhow!(
+                "Commit transaction {commit_tx_hash} for batch {} is not successful",
+                batch_number
+            );
+            return Err(err);
+        }
+
+        let event = self
+            .contract
+            .event("BlockCommit")
+            .context("`BlockCommit` event not found for ZKsync L1 contract")?;
+
+        let commited_batch_info: Option<(H256, H256)> =
+            receipt.logs.into_iter().filter_map(|log| {
+                if log.address != self.diamond_proxy_addr {
+                    tracing::debug!(
+                        "Log address {} does not match diamond proxy address {}, skipping",
+                        log.address,
+                        self.diamond_proxy_addr
+                    );
+                    return None;
+                }
+                let parsed_log = event
+                                        .parse_log_whole(log.into())
+
+                    .ok()?; // Skip logs that are of different event type
+
+                let block_number_from_log = get_param(&parsed_log.params, "batchNumber")
+                        .and_then(ethabi::Token::into_uint)
+                        .and_then(|x| u32::try_from(x).ok())
+                        .map(L1BatchNumber)
+                        .expect("Missing expected `batchNumber` parameter in `BlockCommit` event log");
+
+                if block_number_from_log != batch_number {
+                    tracing::warn!(
+                        "Commit transaction {commit_tx_hash:?} has `BlockCommit` event log with batchNumber={block_number_from_log}, \
+                        but we are checking for batchNumber={batch_number}"
+                    );
```

Comment:
why warn? 
it's kinda normal that we have more than 1 batch in commit tx

Reply:
Commit? Execute yes, but commit is the one with pubdata publishing, its not very normal I think to batch it, but I can change to debug.